lib/subprocess.rb

```
@@ -429,6 +430,11 @@ def communicate(input=nil, timeout_s=nil)
             end
           end
 
+          if ready_r.include?(global_read)
+            drain_fd(global_read)
```

I think we should raise an exception if this returns true as with `self_read`